go/libkb/leveldb.go

```diff
@@ -241,21 +242,25 @@ func (l *LevelDb) isCorrupt(err error) bool {
 	if strings.Contains(err.Error(), "corrupt") {
 		return true
 	}
-
+	// if our db is in a bad state with too many open files also nuke
+	if strings.Contains(strings.ToLower(err.Error()), "too many open files") {
+		return true
+	}
 	return false
 }
 
 func (l *LevelDb) Nuke() (fn string, err error) {
 	l.Lock()
 	// We need to do deferred Unlock here in Nuke rather than delegating to
 	// l.Close() because we'll be re-opening the database later, and it's
-	// necesary to block other doWhileOpenAndNukeIfCorrupted() calls.
+	// necessary to block other doWhileOpenAndNukeIfCorrupted() calls.
 	defer l.Unlock()
 	defer l.G().Trace("LevelDb::Nuke", func() error { return err })()
 
+	// even if we can't close the db try to nuke the files directly
 	err = l.closeLocked()
 	if err != nil {
-		return "", err
+		l.G().Log.Debug("Error closing leveldb %v, attempting nuke anyway", err)
```

Comment:
is this safe?

Reply:
even the close couldn't completely successful i don't think it matters since we blow away all of the files right after this. our local state is valid since we set the `db=nil` even if the close failed: https://github.com/keybase/client/blob/6e033eaa414f7d9fa45b5713cd781076c9fdf287/go/libkb/leveldb.go#L221

Reply:
does this log people out? or only if they restart keybase?

Reply:
doesn't affect their session just the sigchain/team/chat caches